### PR TITLE
Bug 1951158: modify ip to use IP CIDR format

### PR DESCRIPTION
**Pull request by msherif1234**
Signed-off-by: Mohamed Mahmoud <mmahmoud@redhat.com>
modify egress router CNI to allow using CIDR ips for addresses instead of IP

### CODE REVIEW

File: networkoperator/v1/001-egressrouter.crd.yaml-patch [hunk 1]
```diff
@@ -1,8 +1,6 @@
 - op: add
-  path: /spec/versions/name=v1/schema/openAPIV3Schema/properties/spec/properties/addresses/items/properties/ip/anyOf
-  value:
-    - format: ipv4
-    - format: ipv6
+  path: /spec/versions/name=v1/schema/openAPIV3Schema/properties/spec/properties/addresses/items/properties/ip/type
```
Comment (squeed) [inline networkoperator/v1/001-egressrouter.crd.yaml-patch:2]: You can get rid of these patch entries, since string is already correct.

**Comment by squeed** [comment]
One small nit. Unless the OpenAPI understood by the k8s apiserver supports a cidr validator, this is correct. The regex for a ipv4 / ipv6 cidr is like 1600 bytes.

**Comment by msherif1234** [comment]
> One small nit, otherwise OK.
> 
> Unless the OpenAPI understood by the k8s apiserver supports a cidr validator, this is correct. The regex for a ipv4 / ipv6 cidr is like 1600 bytes.

string type is the proper generalization of what we can configure it seems for this feature u can 
x.x.x.x/y or just x.x.x.x and different actions will be needed if u pass in the IP in CIDR fmt vs just IP but both are acceptable 
so we can't verify IP or CIDR only

**Comment by squeed** [comment]
actually wait.. @tssurya I thought the destination needed to be an IP, not a CIDR. Is this right?

**Comment by msherif1234** [comment]
> actually wait.. @tssurya I thought the destination needed to be an IP, not a CIDR. Is this right?

[egress-router-pod](https://docs.openshift.com/container-platform/3.11/admin_guide/managing_networking.html#admin-guide-deploying-an-egress-router-pod)

yeah on the address is CIDR not the egress_destination ?

**Comment by squeed** [comment]
@msherif1234 that's a bug in egress-router-pod -- `-j DNAT --to-destination` doesn't take a CIDR. In this case, the CRD is correct and the pod is wrong.

**Comment by msherif1234** [comment]
> @msherif1234 that's a bug in egress-router-pod -- `-j DNAT --to-destination` doesn't take a CIDR. In this case, the CRD is correct and the pod is wrong.

Yes you are right @squeed I fixed the CNI to parse it as an IP not CIDR so we only left with ip field I updated the PR accordingly

**Pull request merged**